src/vm/eventpipebuffer.cpp

```diff
@@ -54,35 +54,65 @@ bool EventPipeBuffer::WriteEvent(Thread *pThread, EventPipeSession &session, Eve
         NOTHROW;
         GC_NOTRIGGER;
         MODE_ANY;
-        PRECONDITION(pThread != NULL);
         PRECONDITION(((size_t)m_pCurrent % AlignmentSize) == 0);
     }
     CONTRACTL_END;
 
     // Calculate the size of the event.
     unsigned int eventSize = sizeof(EventPipeEventInstance) + payload.GetSize();
+    DWORD osThreadId;
 
     // Make sure we have enough space to write the event.
     if(m_pCurrent + eventSize >= m_pLimit)
     {
         return false;
     }
 
+    if (pThread == NULL)
+    {
+        // For ThreadID just get the Current Thread ID
```

Comment:
I do not think you want to be mixing managed thread IDs and OS thread IDs. The thread ID becomes non-nonsensical once you do that - multiple threads can have same ID, etc.